Source/Urho3D/IO/FileSystem.cpp

```diff
@@ -1599,6 +1677,11 @@ ea::string FileSystem::FindResourcePrefixPath() const
     return EMPTY_STRING;
 }
 
+ea::string ResolvePath(ea::string_view path)
+{
+    return DotDotHelper().Parse(path);
```

Comment:
I would suggest not having extra indirections and making code simpler. Code from `Parse()` should go into this function. `ProcessSegment()` may just be a static function taking two parameters, it also should be right above `ResolvePath()` if it is only used here.

Reply:
Sure... I don't think it is better now to be honest.

Reply:
It has 4 arguments now, and it worth to pack them in the struct. I think the function should be the method of the struct... And the original version would look better.